app/src/main/java/ch/epfl/sdp/blindly/helpers/DatatbaseHelper.kt

```diff
@@ -0,0 +1,206 @@
+            databaseInstance.getReference("messages").child(getConversationId(userId, otherUserId)),
+            userId
+        )
+    }
+
+    /**
+     * The live database for the location
+     *
+     * @param userId the logged-in user id
+     * @param otherUserId the other user's id to share the location with
+     * @return The live database
+     */
+    fun getLocationLiveDatabase(
+        userId: String,
+        otherUserId: String
+    ): LocationLiveDatabase {
+        return LocationLiveDatabase(
+            databaseInstance.getReference("locations").child(getConversationId(userId, otherUserId)),
+            userId
+        )
+    }
+
+    /**
+     * The live database to share the location live
+     *
+     * @constructor
+     * Internal constructor to get an instance see [getLocationLiveDatabase]
+     *
+     * @param dr the firebase database reference
+     * @param userId the logged-in user id
+     */
+    class LocationLiveDatabase internal constructor(dr: DatabaseReference, userId: String) :
+        BlindlyLiveDatabase<BlindlyLatLng>(dr, userId) {
+        override val typeIndicator = object : GenericTypeIndicator<Message<BlindlyLatLng>>() {}
+
+        fun updateLocation(location: BlindlyLatLng) {
+            val newMessage = Message(
+                location,
+                userId
+            )
+            // We keep only one entry
+            dr.child(userId).setValue(newMessage)
+        }
+    }
+
+    /**
+     * The live database for the chat
+     *
+     * @constructor
+     * Internal constructor to get an instance see [getChatLiveDatabase]
+     *
+     * @param dr the firebase database reference
+     * @param userId the logged-in user id
+     */
+    class ChatLiveDatabase internal constructor(dr: DatabaseReference, userId: String) :
+        BlindlyLiveDatabase<String>(dr, userId) {
+        override val typeIndicator = object : GenericTypeIndicator<Message<String>>() {}
+
+        fun sendMessage(message: String) {
+            val newMessage = Message(
+                message,
+                userId
+            )
+            dr.child(newMessage.timestamp.toString()).setValue(newMessage)
+        }
+    }
+
+    /**
+     * A generic live database using [Message] to communicate to other users
+     *
+     * @param T The type contained in messages, see [Message]
+     * @constructor
+     * Internal constructor to create an instance, use one of the public methods to
+     * get an instance
+     *
+     * @param dr the firebase database reference
+     * @param userId the logged-in user id
+     */
+    abstract class BlindlyLiveDatabase<T> internal constructor(protected val dr: DatabaseReference,
+                                                               protected val userId: String
+    ) {
+        // Init at one as most of the time we don't register more than one listener per db
+        private val eventListeners: MutableList<EventListener<T>> = ArrayList(1)
```

Comment:
Magic number... (sorry this is really nitpicking)

Reply:
Well it is just one and doesn't change anything to the functionality. It is just a (very little) performance tuning